### ISSUE TITLE
Another copy of Jenkins CasC "jenkins_user.yaml" will hinder the upgrade of "jenkins.yaml"

### ISSUE THREAD

**Issue by JohnNiang** [kubesphere/ks-devops#359]
If we make some upgrades into `jenkins.yaml`, like bumping builders' image version, people who have already installed KubeSphere 3.2.0 won't obtain those upgrades after KubeSphere upgrade. Because the `jenkins_user.yaml` has been fixed after the first installation.

![image](https://user-images.githubusercontent.com/16865714/141611494-e9f8b854-0178-4a0e-8f1f-ec7c1c52ab01.png)

**Comment by JohnNiang** [comment]
After I upgrade helm chart of ks-devops, the new `jenkins-casc-config` ConfigMap won't be applied. Please see:
![image](https://user-images.githubusercontent.com/16865714/156143225-eb26d15a-2828-4d73-ae11-a015710f7fc1.png)
![image](https://user-images.githubusercontent.com/16865714/156143172-c483f671-0845-4240-9269-6d860571b8b9.png)

### How to upgrade?

0. Enter shell of ks-installer Pod
1. Clone https://github.com/kubesphere-sigs/ks-devops-helm-chart.git from master branch and get in
2. Execute the following command
    ```bash
    helm -n kubesphere-devops-system upgrade devops charts/ks-devops/ --install --reuse-values
    ```
3. See the result

I don't have a solution right now, just let the problem emerge again.

**Comment by LinuxSuRen** [comment]
I don't think this is a bug. Because this is expected according to our design. I prefer to have an enhancement. Considering the team's capacity, I'm going to lower the priority of this issue.

/remove-kind bug
/remove-priority high
/kind feature
